plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -133,48 +146,33 @@ scriptCBORDecoder lv pv =
-fromSerialisedScript :: forall e m. (AsScriptDecodeError e, MonadError e m)
+-- | The deserialization from a serialised script into a `ScriptForEvaluation`,
+-- ready to be evaluated on-chain.
+-- Called inside phase-1 validation (i.e., deserialisation error is a phase-1 error).
+deserialiseScript :: forall m. (MonadError ScriptDecodeError m)
                      => PlutusLedgerLanguage -- ^ the Plutus ledger language of the script.
                      -> ProtocolVersion -- ^ which protocol version the script was submitted in.
                      -> SerialisedScript -- ^ the script to deserialise.
-                     -> m ScriptForExecution
-fromSerialisedScript lv currentPv sScript = do
+                     -> m ScriptForEvaluation
+deserialiseScript ll pv sScript = do
     -- check that the ledger language version is available
-    let llIntroPv = ledgerLanguageIntroducedIn lv
-    unless (llIntroPv <= currentPv)  $
-        throwing _ScriptDecodeError $ LedgerLanguageNotAvailableError lv llIntroPv currentPv
+    let llIntroPv = ledgerLanguageIntroducedIn ll
+    unless (llIntroPv <= pv)  $
+        throwing _ScriptDecodeError $ LedgerLanguageNotAvailableError ll llIntroPv pv
 
-    (remderBS, script@(ScriptForExecution (UPLC.Program{}))) <- deserialiseSScript sScript
-    when (lv /= PlutusV1 && lv /= PlutusV2 && remderBS /= mempty) $
+    (remderBS, dScript@(ScriptNamedDeBruijn (UPLC.Program{}))) <- deserialiseSScript sScript
+    when (ll /= PlutusV1 && ll /= PlutusV2 && remderBS /= mempty) $
         throwing _ScriptDecodeError $ RemainderError remderBS
 
-    pure script
+    pure $ UnsafeScriptForEvaluation sScript dScript
   where
-    deserialiseSScript :: SerialisedScript -> m (BSL.ByteString, ScriptForExecution)
+    deserialiseSScript :: SerialisedScript -> m (BSL.ByteString, ScriptNamedDeBruijn)
     deserialiseSScript = fromShort
                        >>> fromStrict
-                       >>> CBOR.deserialiseFromBytes (scriptCBORDecoder lv currentPv)
+                       >>> CBOR.deserialiseFromBytes (scriptCBORDecoder ll pv)
                        -- lift the underlying cbor error to our custom error
                        >>> either (throwing _ScriptDecodeError . toScripDecodeError) pure
 
     -- turn a cborg failure to our own error type
     toScripDecodeError :: CBOR.DeserialiseFailure -> ScriptDecodeError
     toScripDecodeError = CBORDeserialiseError . CBOR.Extras.readDeserialiseFailureInfo
-
-
-{-| Check if a 'Script' is "valid" according to a protocol version. At the moment this means "deserialises correctly", which in particular
-implies that it is (almost certainly) an encoded script and the script does not mention any builtins unavailable in the given protocol version.
-
-Note: Parameterized over the ledger-plutus-version since the builtins allowed (during decoding) differs.
--}
-assertScriptWellFormed :: MonadError ScriptDecodeError m
```

Comment:
We agreed that we were going to leave the old interface in place, so that we can update without the ledger team having to update immediately.

So either we should leave this function in, or we should be prepared to re-create it on the ledger side when we update them. Not sure which is better.